R-package/R/lgb.Booster.R

```diff
@@ -307,6 +307,46 @@ Booster <- R6::R6Class(
 
     },
 
+    # Number of models (~trees) per iteration
+    num_models_per_iter = function() {
+
+      self$restore_handle()
+
+      models_per_iter <- 1L
+      .Call(
+        LGBM_BoosterNumModelPerIteration_R
+        , private$handle
+        , models_per_iter
+      )
+      return(models_per_iter)
+
+    },
+
+    # Total number of models (~trees)
+    num_total_models = function() {
```

Comment:
I think this should be called `num_trees()` instead, since people might be familiar with that from the Python package:

https://github.com/microsoft/LightGBM/blob/fd8f79944955cf83fcefb9ed598b9e0bcd07ad09/python-package/lightgbm/basic.py#L4241

And maybe also change `num_models_per_iter()` to `num_trees_per_iter()`?

The language "model" as a synonym for "tree" isn't really used anywhere else in the public API of the R package. Except for one old demo, all the references to "model" in the R package refer to the entire ensemble:

https://github.com/search?q=repo%3Amicrosoft%2FLightGBM+model+language%3AR&type=code&l=R&p=1

It looks to me like that is how "model" is used in `{xgboost}` as well: https://github.com/dmlc/xgboost/blob/master/R-package/R/xgb.Booster.R.

I think that internal consistency across the R API is more important than consistency between the R-package method names and the C API function names, in this case. I don't expect most users of the R package to also have any familiarity with its C API.

Reply:
I was thinking about this as well. Let's change to "trees" and the shorter `num_trees()`.